### DISCUSSION TITLE
Admin console login call to get token takes more time compare to legacy keycloak

### DISCUSSION THREAD

**Discussion by sreekesh93** [keycloak/keycloak#15187] [Archive]
Existing Setup :
I have deployed the keycloak legacy version 18.0.0 initially, In this setup there are close to 1000 realms created. 
on top of this we have enabled keycloak quarkus 18.0.0 version and migration was success full

Scenario tested:
We tried to login Admin console page with legacy version we could note that token fetch call takes 2s max, after applying Quarkus version of keycloak the token generation call nearly takes 40s to fetch token while login to admin console

Is there any performance issue with quarkus version of keycloak? If not are we missing any config please suggest us some techinques to address the issue.

**Comment by sreekesh93** [comment]
@stianst / @pedroigor  can you please share your thoughts?

**Comment by kkcmadhu** [comment]
i too face the same issue, i realised that across most calls (token,logout, search user, roles etc etc) there is at the least about  10 to 15%  increase in latency ..my hunch is the slowness got introduced somewhere  in keycloak 16 or keycloak 17..

i am using legacy store with quarkus runtime for  keycloak 17

**Reply by sreekesh93** [reply]
New store is not yet production ready is what I understood so far...

**Reply by kkcmadhu** [reply]
yes not yet ! like you in old store too, i see slowness between keycloak 15 and 17.

**Reply by sreekesh93** [reply]
I would suggest to enable the ispn cache config it has reduced the latency to a manageable extend

**Comment by ronycha** [comment]
I saw the same issue like above mentioned by @kkcmadhu in Keycloak17 wildfly and Keyloak 1903 quarkus with old store.
